js/private/js_library.bzl

```diff
@@ -19,106 +18,124 @@ js_library(
     ],
 )
 ```
+
+| This is similar to [`py_library`](https://docs.bazel.build/versions/main/be/python.html#py_library) which depends on
+| Python sources and provides a `PyInfo`.
 """
 
-load("@aspect_bazel_lib//lib:copy_to_bin.bzl", "copy_files_to_bin_actions")
-load("@rules_nodejs//nodejs:providers.bzl", "DeclarationInfo", "declaration_info")
+load(":js_info.bzl", "JsInfo")
+load(":js_library_helpers.bzl", "gather_npm_linked_packages", "gather_npm_package_stores", "gather_runfiles", "gather_sources_and_declarations", "gather_transitive_declarations", "gather_transitive_sources")
+
+_DOC = """A library of JavaScript sources. Provides JsInfo, the primary provider used in rules_js
+and derivate rule sets.
+
+Declaration files are handled separately from sources since they are generally not needed at
+runtime and build rules, such as ts_project, are optimal in their build graph if they only depend
+on declarations from 'deps' since these they don't need the JavaScript source files from deps to
+typecheck.
 
-_DOC = """Copies all sources to the output tree and expose some files with DeclarationInfo.
+Linked npm dependences are also handled separately from sources since not all rules require them and it
+is optimal for these rules to not depend on them in the build graph.
 
-Can be used as a dep for rules that expect a DeclarationInfo such as ts_project."""
+NB: 'js_library' copies all source files to the output tree before providing them in JsInfo. See
+https://github.com/aspect-build/rules_js/tree/67c561e08fa8baaf6fbfe00094fa0d757159e1ef#running-nodejs-programs
```

Comment:
https://github.com/aspect-build/rules_js/tree/67c561e08fa8baaf6fbfe00094fa0d757159e1ef/docs#javascript might be a better link target

Reply:
✅